cli/tests/unit/dispatch_json_test.ts

```diff
@@ -35,8 +35,9 @@ unitTest(function invalidPromiseId(): void {
     },
   };
   const argsText = JSON.stringify(argsObj);
-  const argsBuf = new TextEncoder().encode(argsText);
-  const resBuf = Deno.core.send(opId, argsBuf);
+  const reqText = JSON.stringify([promiseId, argsText]);
```

Comment:
I am not in favor of this double JSON encoding of args. What is the reason for this?

Reply:
The previous implementation threw a deno-specific `type_error` when the promiseId was invalid. Serde does not provide an interface for easily adding your own error types to decoding, so my best solution was to decode promiseId and the args separately. Without hand-writing a manual decode routine, this was the best way I could think of to separate the two decodes.

Reply:
Can I suggest you put the raw bytes representing the `promiseId` i64 into the first 8 bytes of the sent Uint8Array?The remaining bytes after the 8 raw bytes for the promise ID could then be the JSON bytes. That way we avoid this double JSON encode/decode.

Reply:
That's a much better solution. In that case, would this test still be necessary?

Reply:
Yeah, for example if the sent array only has 4 bytes.